packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1223,13 +1232,26 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
   void _listenToStream() {
     if (_isListeningToStream)
       return;
+
     _imageStream!.addListener(_getListener());
+    if (_passiveListener != null) {
+      assert(_imageStream?.completer != null);
+     _imageStream!.completer!.removePassiveListener(_passiveListener!);
+     _passiveListener = null;
+    }
+
     _isListeningToStream = true;
   }
 
-  void _stopListeningToStream() {
+  void _stopListeningToStream([bool createPassiveListener = false]) {
     if (!_isListeningToStream)
       return;
+
+    if (createPassiveListener && _passiveListener == null && _imageStream?.completer != null) {
```

Comment:
A doc comment somewhere explaining why stopping to listen may require a passive listener would be helpful.

Reply:
Done